app/models/email_configuration.rb

```diff
@@ -18,7 +18,7 @@ class EmailConfiguration < ApplicationRecord
   belongs_to :account, optional: false
   validate :account_is_provider
 
-  before_save :set_tenant_id, on: :create
+  before_create :set_tenant_id
```

Comment:
need to change this? I remember some weird issues reported for some rails versions of difference between `before_save` and `before_create` so prefer to keep as is if we have no important reason

Reply:
You are right @akostadinov 

I've looked into it, and the "weird issues" is basically that `before _save` just ignores the `on:` option: https://github.com/rails/rails/issues/17622

This was "fixed" in Rails 6.0 by disallowing the `on:` options: https://github.com/rails/rails/pull/30919

Now it throws
```
Uncaught exception: Unknown key: :on. Valid keys are: :if, :unless, :prepend
```

So yeah, this change is required, the question is - should we just keep `before_save` (because I guess that would be keeping the existing behavior), or switch to `before_create` (which I assume was the intended - although not working - behavior)?

Reply:
In this particular case I think we can use just `before_save` - in case we have some older models that did not have `tenant_id` initially, they will still get it on update.

Reply:
we should use `before_create` because `before_save` would call `set_tenant_id` also when `#update`. Is `before_create` not working? Why?

Reply:
@jlledom what I meant is that right now I think `before_save :set_tenant_id` is the equivalent of `before_save :set_tenant_id, on: :create`

So, by keeping `before_save` we're just maintaining the existing logic.

And my point is that I think it's OK if we call this method on `#update` also, as it will do nothing if `tenant_id` already is set.

Reply:
I changed this to `before_save`